fe/fe-core/src/main/java/com/starrocks/authentication/AuthenticationMgr.java

```diff
@@ -248,7 +253,75 @@ private UserIdentity checkPasswordForNative(
     }
 
     public UserIdentity checkPassword(String remoteUser, String remoteHost, byte[] remotePasswd, byte[] randomString) {
-        return checkPasswordForNative(remoteUser, remoteHost, remotePasswd, randomString);
+        UserIdentity authenticatedUser = null;
+        String userAndHost = remoteUser.concat("@").concat(remoteHost);
+        for (Map.Entry<String, Long> attempt : FAIL_TIME.entrySet()) {
+            if (((System.currentTimeMillis() - attempt.getValue()) / 1000) > Config.password_lock_interval_seconds) {
+                if (!isUserLocked(attempt.getKey())) {
+                    clearFailedAttemptRecords(attempt.getKey());
+                } else {
+                    if (getRemainingLockedTime(attempt.getKey()) <= 0) {
+                        clearFailedAttemptRecords(attempt.getKey());
+                    }
+                }
+            }
+        }
+        if (!allowLoginAttempt(userAndHost)) {
+            return null;
+        }
+        authenticatedUser = checkPasswordForNative(remoteUser, remoteHost, remotePasswd, randomString);
+        if (Config.max_failed_login_attempts >= 0) {
```

Comment:
is it possible to prompt to user, the account will be locked for next xx fail attempts